Marlin/Marlin_main.cpp

```diff
@@ -1287,7 +1288,25 @@ static void update_software_endstops(AxisEnum axis) {
   {
     sw_endstop_min[axis] = base_min_pos(axis) + offs;
     sw_endstop_max[axis] = base_max_pos(axis) + offs;
+
+    #if ENABLED(AUTO_BED_LEVELING_FEATURE)
+      if (axis == Z_AXIS) {
```

Comment:
As far as I can tell, this code should accomplish the same thing as it did when it was in `clamp_to_software_endstops`. If it was doing the right thing in that function, then I don't see why it should apply any differently here. I don't know whether it matters which way the machine homes Z, since these "negative offsets" are meant to apply to the Z lower limit, and I'm not sure the Z homing direction matters for this extra consideration.